### PR TITLE
Add reflection unit tests

### PR DESCRIPTION
**Pull request by Marczewka**
I created ReflectionTest Utility Class that makes unit tests for all Reflection class functions and testClasses class that stores classes for testing. I also deleted all TO DO comments in Reflection class.

### CODE REVIEW

File: src/test/java/com/github/splendor_mobile_game/websocket/utils/reflection/testClass.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package com.github.splendor_mobile_game.websocket.utils.reflection;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+public class testClass {
```
Comment (kestivvi) [inline src/test/java/com/github/splendor_mobile_game/websocket/utils/reflection/testClass.java:9]: Add simple one line java doc for this class.

File: src/test/java/com/github/splendor_mobile_game/websocket/utils/reflection/ReflectionTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.lang.reflect.Constructor;
+import java.util.List;
+
+public class ReflectionTest {
```
Comment (kestivvi) [inline src/test/java/com/github/splendor_mobile_game/websocket/utils/reflection/ReflectionTest.java:20]: Add simple one line java doc for this class.

**Review by kestivvi — CHANGES_REQUESTED** [review]

**Comment by kestivvi** [comment]
Awesome ✨✨💎💎

**Review by kestivvi — APPROVED** [review]
Brilliant unit tests. Others can learn from you 🤩

**Review by agatamrnw — APPROVED** [review]

**Pull request merged**